pkg/asset/installconfig/installconfig.go

```diff
@@ -157,6 +157,13 @@ func (a *InstallConfig) finish(filename string) error {
 	}
 	if a.Config.Azure != nil {
 		a.Azure = icazure.NewMetadata(a.Config.Azure.CloudName, a.Config.Azure.ARMEndpoint)
+		if a.Config.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet != nil && a.Config.Platform.Azure.DefaultMachinePlatform.OSDisk.DiskEncryptionSet.SubscriptionID == "" {
```

Comment:
This will fail if there is no `defaultMachinePlatform` specified in the install-config.yaml (or `DefaultMachinePlatform.OSDisk`):
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x3486976]

goroutine 1 [running]:
github.com/openshift/installer/pkg/asset/installconfig.(*InstallConfig).finish(0xc00073fdc0, {0x7be442d, 0x13})
	~/prjs/upstream/openshift/installer/pkg/asset/installconfig/installconfig.go:160 +0x3b6
github.com/openshift/installer/pkg/asset/installconfig.(*InstallConfig).Load(0xc00073fdc0, {0x1f640c60?, 0xc00073cbc0?})
	~/prjs/upstream/openshift/installer/pkg/asset/installconfig/installconfig.go:120 +0x5b
github.com/openshift/installer/pkg/asset/store.(*storeImpl).load(0xc0008a7b60, {0x1f648930, 0xc00073eb40}, {0x7ba4cf0, 0x2})
	~/prjs/upstream/openshift/installer/pkg/asset/store/store.go:263 +0x2b2
github.com/openshift/installer/pkg/asset/store.(*storeImpl).load(0xc0008a7b60, {0x7f54f0c6f718, 0x241959d0}, {0x0, 0x0})
	~/prjs/upstream/openshift/installer/pkg/asset/store/store.go:246 +0xc05
github.com/openshift/installer/pkg/asset/store.(*storeImpl).fetch(0xc0008a7b60, {0x7f54f0c6f718, 0x241959d0}, {0x0, 0x0})
	~/prjs/upstream/openshift/installer/pkg/asset/store/store.go:200 +0x1a9
github.com/openshift/installer/pkg/asset/store.(*storeImpl).Fetch(0x7ffed232f883?, {0x7f54f0c6f718, 0x241959d0}, {0x241768c0, 0x6, 0x6})
	~/prjs/upstream/openshift/installer/pkg/asset/store/store.go:76 +0x48
main.runTargetCmd.func1({0x7ffed232f883, 0x2})
	~/prjs/upstream/openshift/installer/cmd/openshift-install/create.go:260 +0x125
main.runTargetCmd.func2(0x2417dee0?, {0xc00073e740?, 0x4?, 0x4?})
	~/prjs/upstream/openshift/installer/cmd/openshift-install/create.go:290 +0xe7
github.com/spf13/cobra.(*Command).execute(0x2417dee0, {0xc00073e700, 0x4, 0x4})
	~/prjs/upstream/openshift/installer/vendor/github.com/spf13/cobra/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xc0004bac00)
	~/prjs/upstream/openshift/installer/vendor/github.com/spf13/cobra/command.go:1040 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	~/prjs/upstream/openshift/installer/vendor/github.com/spf13/cobra/command.go:968
main.installerMain()
	~/prjs/upstream/openshift/installer/cmd/openshift-install/main.go:61 +0x2b0
main.main()
	~/prjs/upstream/openshift/installer/cmd/openshift-install/main.go:38 +0xff
```

Reply:
Fixed.